parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -342,51 +357,89 @@ pub mod pallet {
 		}
 
 		#[pallet::call_index(3)]
-		#[pallet::weight({1000})]
+		#[pallet::weight(T::WeightInfo::block_bridge())]
 		#[transactional]
-		pub fn unblock_bridge(origin: OriginFor<T>) -> DispatchResult {
+		pub fn block_bridge(origin: OriginFor<T>) -> DispatchResult {
 			let _sender = ensure_root(origin)?;
 
-			<Blocked<T>>::set(false);
+			<Blocked<T>>::set(true);
+
+			log::info!(target: "ethereum-beacon-client","💫 bridge emergency blocked from governance.");
+
+			Self::deposit_event(Event::Deactivated);
+
+			Ok(())
+		}
+
+		#[pallet::call_index(4)]
+		#[pallet::weight(T::WeightInfo::begin_recovery())]
+		#[transactional]
+		pub fn begin_recovery(
+			origin: OriginFor<T>,
+			check_points: Vec<CheckPointRoot>,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			for check_point in check_points {
+				<AuthorizedCheckPoints<T>>::insert(check_point, true)
+			}
+			Ok(())
+		}
 
-			log::info!(target: "ethereum-beacon-client","💫 syncing bridge from governance provided checkpoint.");
+		#[pallet::call_index(5)]
+		#[pallet::weight(T::WeightInfo::sync_recovery())]
+		#[transactional]
+		pub fn sync_recovery(
+			origin: OriginFor<T>,
+			checkpoint: CheckpointSyncOf<T>,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+			let checkpoint_root: CheckPointRoot = hash_tree_root_check_point(checkpoint.clone())
+				.map_err(|_| Error::<T>::CheckpointHashTreeRootFailed)?
+				.into();
+			ensure!(
+				<AuthorizedCheckPoints<T>>::contains_key(checkpoint_root),
+				Error::<T>::CheckpointInvalid
+			);
+			Self::process_checkpoint_sync(checkpoint)?;
+			Ok(())
+		}
 
+		#[pallet::call_index(6)]
+		#[pallet::weight(T::WeightInfo::unblock_bridge())]
+		#[transactional]
+		pub fn unblock_bridge(origin: OriginFor<T>) -> DispatchResult {
+			ensure_root(origin)?;
+			<Blocked<T>>::set(false);
+			Self::deposit_event(Event::Activated);
 			Ok(())
 		}
 	}
 
 	impl<T: Config> Pallet<T> {
 		fn process_initial_sync(initial_sync: InitialSyncOf<T>) -> DispatchResult {
```

Comment:
Discussion: We can unify the `process_initial_sync` and `unblock_bridge` code paths. Instead of having an initial sync step, we could start the bridge in a `Deactivated` state by default and then use the `unblock_bridge` extrinsic to bring it into an Activated state. This will simplify the deployment process as we will not have to set `initial_sync` field in the beacon client genesis state.

Reply:
Good point!  Some changes accordingly.
https://github.com/Snowfork/snowbridge/pull/795/commits/4b1279a84029f407945df500c767d56b6b52deb3

Reply:
I wonder if the methods shouldn't then mirror the language used, for example, rename `block_bridge` to `deactivate` and `unblock_bridge` to `activate`?